lib/streamlit/elements/image.py

```diff
@@ -489,7 +526,16 @@ def marshall_images(
 
             # Following regex allows svg image files to start either via a "<?xml...>" tag eventually followed by a "<svg...>" tag or directly starting with a "<svg>" tag
             if re.search(r"(^\s?(<\?xml[\s\S]*<svg\s)|^\s?<svg\s)", image):
-                proto_img.markup = f"data:image/svg+xml,{image}"
+                if image.find("xmlns:xlink") != -1:
+                    proto_img.markup = f"data:image/svg+xml,{image}"
```

Comment:
Does the `width` parameter of `st.image` work in this case correctly? I would assume that if an SVG is passed that contains an `xlmns` tag, the width parameter is not working.

Reply:
Btw. why don't we just always use the `img` tag to render the SVG instead of doing this? 

https://github.com/streamlit/streamlit/blob/5edf0aba3a828082457427e7c5f6d7f5f0943f2f/frontend/src/components/elements/ImageList/ImageList.tsx#L103

Reply:
Actually it's not I've not managed to scale SVG's which are linked externally (they have `xmlns:xlink` attribute) or they do not have both `viewBox` and `width` attribute set. (They will be rendered as they were previously, so nothing will change in this cases.)

Reply:
> Btw. why don't we just always use the `img` tag to render the SVG instead of doing this?
> 
> https://github.com/streamlit/streamlit/blob/5edf0aba3a828082457427e7c5f6d7f5f0943f2f/frontend/src/components/elements/ImageList/ImageList.tsx#L103

When we don't have both `viewBox` and `width` and `height` set the only way to render it properly is to use markup instead src of an image tag.

Same goes for `xmlns:xlink` which are `svgs` containing links to external images, they seem to only be rendered properly as separate tags instead of `src` of `img` HTML tag.

Reply:
Is `xmlns:xlink` related to `xlink:href`? If so, this seems to be a deprecated feature and we might not need to support it, see here: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href